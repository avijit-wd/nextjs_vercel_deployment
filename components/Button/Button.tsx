import styled from "@emotion/styled";
import { css, SerializedStyles } from "@emotion/react";
import { MouseEvent } from "react";

export type Color = "primary" | "secondary" | "danger" | "warning";

export type ButtonProps = {
  children: string;
  color?: Color;
  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
};

export const getColors = (color?: Color): SerializedStyles => {
  switch (color) {
    case "primary":
      return css`
        background-color: violet;
        color: "white";
      `;
    case "secondary":
      return css`
        background-color: green;
        color: "white";
      `;
    case "danger":
      return css`
        background-color: red;
        color: "white";
      `;
    case "warning":
      return css`
        background-color: yellow;
        color: "white";
      `;
    default:
      return css``;
  }
};

export const Button = styled.button<ButtonProps>`
  all: unset;
  display: flex;
  align-items: center;
  justify-self: center;
  justify-content: center;
  user-select: none;
  cursor: pointer;
  font-size: 1.6rem;
  width: 15rem;
  height: 4rem;
  border-radius: 1rem;
  transition: all 0.4s ease;
  ${({ color }) => getColors(color)};
  &:hover {
    opacity: 0.9;
  }
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);

  &:active {
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.1);
  }
`;

Button.defaultProps = {
  color: "primary",
};
