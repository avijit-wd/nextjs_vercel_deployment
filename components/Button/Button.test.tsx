import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "./Button";

describe("Button test case", () => {
  it("Render check", () => {
    const mockFn = jest.fn();
    const { asFragment } = render(<Button onClick={mockFn}>Button</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Check onClick callback", () => {
    const onClick = jest.fn();

    render(<Button onClick={onClick}>Button</Button>);

    const element = screen.getByRole("button");

    userEvent.click(element);
  });
});
