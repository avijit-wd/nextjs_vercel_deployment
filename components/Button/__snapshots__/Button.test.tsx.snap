// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test case Render check 1`] = `
<DocumentFragment>
  <button
    class="css-i7rjuy"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;
